codec_test.go

```diff
@@ -105,6 +105,40 @@ func TestV2Codec(t *testing.T) {
 	}
 }
 
+func Test_setting_withDecodeOnly_disables_encoding(t *testing.T) {
+	d := &memory.Driver{}
+	srv := httptest.NewServer(server.NewHttpHandler(d))
+	defer srv.Close()
+	decodeOnlyCodec := setUpWithServer(t, "v2", srv, true)
+
+	defaultCodec := setUpWithServer(t, "v2", srv, false)
+
+	payload := common.Payload{
+		Metadata: map[string][]byte{
+			"foo":                     []byte("bar"),
+			"baz":                     []byte("qux"),
+			"remote-codec/key-prefix": []byte("1234"),
+		},
+		Data: []byte("this is a longer message blah blah blah blah blah blah blah"),
+	}
+
+	resp1, err := decodeOnlyCodec.Encode([]*common.Payload{&payload})
+	if err != nil {
+		require.NoError(t, err)
+	}
+	require.Equal(t, &payload, resp1[0])
+
+	resp2, err := defaultCodec.Encode([]*common.Payload{&payload})
+	if err != nil {
+		require.NoError(t, err)
+	}
```

Comment:
I think it would be even more clear, if you add another assert:

```
require.Equal(t, &payload, resp2[0])
```

That shows that the default codec indeed returns a different response. You can also do a `NotEqual` of the responses.